cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/distributed/DistributedProgramRunner.java

```diff
@@ -371,6 +377,30 @@ public final ProgramController run(final Program program, ProgramOptions oldOpti
     }
   }
 
+  /**
+   * For programs initiated by a tethered peer, load additional resources that were saved locally based by runId
+   */
+  private void loadLocalResources(Program program, CConfiguration cConf, String runId)
+    throws URISyntaxException, IOException {
+    File tmpDir = new File(cConf.get(Constants.CFG_LOCAL_DATA_DIR),
+                           cConf.get(Constants.AppFabric.TEMP_DIR)).getAbsoluteFile();
+    for (File file : DirUtils.listFiles(new File(tmpDir, runId))) {
+      if (file.getName().equals(LOGBACK_FILE_NAME)) {
+        // replace existing logback.xml file
+        URI logbackURI = getLogBackURI(program);
+        if (logbackURI != null) {
+          try (InputStream input = logbackURI.toURL().openStream()) {
+            Files.copy(input, file.toPath(), StandardCopyOption.REPLACE_EXISTING);
+          }
+        }
+      } else if (file.getName().equals("cdap-tether.xml")) {
+        // add additional cConf entries
+        cConf.addResource(file.toURI().toString());
```

Comment:
Can there be conflicting entries in cConf and cdap-tether.xml?

Reply:
yes, but we want the cdap-tether entries to override, since we want to use the cConf from the tethered peer when possible

Reply:
Not sure what exactly is in cdap-tether, but wondering if it should replace the cconf or just override some of it. 

If there are properties in the cconf that don't exist in cdap-tether, should they be present?

Reply:
it contains some runtime related cConf entries from the tethered peer that we'd want to use to override in this instance. https://github.com/cdapio/cdap/blob/develop/cdap-app-fabric/src/main/java/io/cdap/cdap/internal/tethering/runtime/spi/runtimejob/TetheringRuntimeJobManager.java#L67-L68 

I don't think we wanted to completely replace this cConf since 1) the select cConf properties would not need to be overridden if this instance was starting a program not initiated by the tethered peer and 2) the versions could be different between this instance and the tethered one, so we may want to keep the properties that are present in this cConf and not in cdap-tether